addons/web/static/src/js/control_panel/search_utils.js

```diff
@@ -108,42 +112,366 @@ odoo.define('web.searchUtils', function (require) {
-    const TIME_RANGE_OPTIONS = {
-        last_7_days: { description: _lt("Last 7 Days"), id: 'last_7_days', groupNumber: 1 },
-        last_30_days: { description: _lt("Last 30 Days"), id: 'last_30_days', groupNumber: 1 },
-        last_365_days: { description: _lt("Last 365 Days"), id: 'last_365_days', groupNumber: 1 },
-        last_5_years: { description: _lt("Last 5 Years"), id: 'last_5_years', groupNumber: 1 },
-        today: { description: _lt("Today"), id: 'today', groupNumber: 2 },
-        this_week: { description: _lt("This Week"), id: 'this_week', groupNumber: 2 },
-        this_month: { description: _lt("This Month"), id: 'this_month', groupNumber: 2 },
-        this_quarter: { description: _lt("This Quarter"), id: 'this_quarter', groupNumber: 2 },
-        this_year: { description: _lt("This Year"), id: 'this_year', groupNumber: 2 },
-        yesterday: { description: _lt("Yesterday"), id: 'yesterday', groupNumber: 3 },
-        last_week: { description: _lt("Last Week"), id: 'last_week', groupNumber: 3 },
-        last_month: { description: _lt("Last Month"), id: 'last_month', groupNumber: 3 },
-        last_quarter: { description: _lt("Last Quarter"), id: 'last_quarter', groupNumber: 3 },
-        last_year: { description: _lt("Last Year"), id: 'last_year', groupNumber: 3 },
-    };
-    const COMPARISON_TIME_RANGE_OPTIONS = {
+    // Comparison menu parameters
+    const COMPARISON_OPTIONS = {
         previous_period: { description: _lt("Previous Period"), id: 'previous_period' },
-        previous_year: { description: _lt("Previous Year"), id: 'previous_year' },
+        previous_year: { description: _lt("Previous Year"), id: 'previous_year', addParam: { years: -1 } },
+    };
+    const COMPARISON_FACTORS = {
+        year: 1,
+        quarter: 4,
+        month: 12,
+    };
+    const COVERED_MONTHS = {
+        1: [0, 1, 2],
+        2: [3, 4, 5],
+        3: [6, 7, 8],
+        4: [9, 10, 11],
     };
 
     // Search bar
     const FACET_ICONS = {
         filter: 'fa-filter',
         groupBy: 'fa-bars',
         favorite: 'fa-star',
-        timeRange: 'fa-calendar',
+        comparison: 'fa-adjust',
     };
 
+    /**
+     * Construct the string representation of a domain and its description. The domain is of the
+     * form ['|',..., '|', d_1,..., d_n] where d_i is a time range of the form
+     * ['&', [fieldName, >=, leftBound_i], [fieldName, <=, rightBound_i]] where leftBound_i and
+     * rightBound_i are date/datetime computed accordingly to the given options and reference moment
+     * (@see constructRange).
+     * @param {Moment} referenceMoment
+     * @param {string} fieldName
+     * @param {string} fieldType
+     * @param {string[]} selectedOptionGeneratorIds
+     * @param {string} [comparisonOptionId]
+     * @returns {{ domain: string, description: string }}
+     */
+    function constructDomain(referenceMoment, fieldName, fieldType, selectedOptionGeneratorIds, comparisonOptionId) {
```

Comment:
this is specific to date fields right? It should maybe be indicated in the function name

Reply:
Done